akka-stream/src/main/scala/akka/stream/Shape.scala

```diff
@@ -270,7 +270,7 @@ final case class BidiShape[-In1, +Out1, -In2, +Out2](in1: Inlet[In1 @uncheckedVa
   override def copyFromPorts(inlets: immutable.Seq[Inlet[_]], outlets: immutable.Seq[Outlet[_]]): Shape = {
     require(inlets.size == 2, s"proposed inlets [${inlets.mkString(", ")}] do not fit BidiShape")
     require(outlets.size == 2, s"proposed outlets [${outlets.mkString(", ")}] do not fit BidiShape")
-    BidiShape(inlets(0), outlets(0), inlets(1), outlets(1))
+    BidiShape(inlets.head, outlets.head, inlets(1), outlets(1))
```

Comment:
For immutable.List:

~~~scala
override def drop(n: Int): List[A] = {
    var these = this
    var count = n
    while (!these.isEmpty && count > 0) {
      these = these.tail
      count -= 1
    }
    these
  }
~~~

so drop(0) == this, then this.head? (I mean cheap as no allocations)

And for vector it is:

~~~scala
def apply(index: Int): A = {
    val idx = checkRangeConvert(index)
    getElem(idx, idx ^ focus)
  }
~~~

(note that `head` on Vector is)

~~~scala
  override /*IterableLike*/ def head: A = {
    if (isEmpty) throw new UnsupportedOperationException("empty.head")
    apply(0)
  }
~~~